bindings/core/src/method/secret_manager.rs

```diff
@@ -19,7 +19,22 @@ use crate::OmittedDebug;
 #[serde(tag = "name", content = "data", rename_all = "camelCase")]
 #[non_exhaustive]
 pub enum SecretManagerMethod {
-    /// Generate Ed25519 addresses.
+    // TODO: Should we refactor `GetAddressesOptions` so we can use it for this method as well?
+    /// Generate a single Ed25519 address.
+    #[serde(rename_all = "camelCase")]
+    GenerateEd25519Address {
```

Comment:
This is already doing https://github.com/iotaledger/iota-sdk/issues/2144 ?

Reply:
Also wondering, what was the motivation to put this here instead of just having it on the JS and Python bindings side where it calls GenerateEd25519Addresses and just returns the first address from the returned Vec? Params you could also change for the functions on the bindings side

Reply:
In parts yes, the issue also addresses whether we should have `Bech32` in the name. At first I thought this should be addressed in a different PR, but I feel like it's fine to do it in this PR already (except for the rename).

Reply:
> Also wondering, what was the motivation to put this here instead of just having it on the JS and Python bindings side where it calls GenerateEd25519Addresses and just returns the first address from the returned Vec? Params you could also change for the functions on the bindings side

Yeah, this would also have been possible. But we had this method already as a wallet method, so it basically just moved.

Reply:
But don't you think this would be better to change it?

Reply:
Not sure if I follow, my question is if you think that `GenerateEd25519Address` in this file is better than just having `GenerateEd25519Addresses` and calling this in `generateEd25519Address()` on the JS side
So for users they still would have both functions as right now, no difference there, only difference would be that we have one less method in this file

Reply:
> more changes than needed like this just often require more discussion/even more changes and delay the merging of PRs...

that's unfortunately true, but I prefer that over merging things before we're happy with it.

Reply:
The bindings secret manager methods have not really anything to do with the wallet? To me this has nothing to do with quality

Reply:
> Not sure if I follow, my question is if you think that `GenerateEd25519Address` in this file is better than just having `GenerateEd25519Addresses` and calling this in `generateEd25519Address()` on the JS side So for users they still would have both functions as right now, no difference there, only difference would be that we have one less method in this file

Yeah, I got what you were saying. We can do that if you think that's better.

For context: My reasoning was that we might be able to deprecate `generateEd25519Addresses` because the functionality of generating multiple at once for a single private key is not that relevant anymore? Maybe I am wrong but the "plural" version seems like a remnant from older IOTA times, but I am not 100% certain :shrug: You probably know better :smile:

Reply:
tbh ... I am fine with removing the single generation version, but strictly speaking this isn't what the PR is about. Just removing that method alltogether is not okay if it existed before and could be moved to the secret manager methods.

Reply:
k ... I'll change it and make it a separate PR but as a matter of fact that method already existed :trollface:

Reply:
`generateEd25519Addresses` can also be used to generate a single address :shrug: 

> functionality of generating multiple at once for a single private key is not that relevant anymore

For a single wallet definitely not, multiple addresses one could still generate with a loop in a less efficient way from bindings side, personally I would actually be fine with only having the single address version
But again, I don't think this should be done in this PR as it already has 40 changed files and the SDK works perfectly fine without changing this as well

Reply:
Changed